r/tests/testthat/test-dplyr.R

```diff
@@ -118,6 +119,15 @@ test_that("filter() with NAs in selection", {
   )
 })
 
+test_that("Filter returning an empty Table should not segfault (ARROW-8354)", {
+  expect_dplyr_error(
```

Comment:
This should not error: it should return a 0-row data.frame:

```r
filter(mtcars,FALSE)
##  [1] mpg  cyl  disp hp   drat wt   qsec vs   am   gear carb
## <0 rows> (or 0-length row.names)
```

I'll pull the branch and try to special-case that.

Reply:
@fsaintjacques I did this in https://github.com/apache/arrow/pull/6871/commits/f93f08c45956f057aad3c019fea0a705b5cba544.